src/TextField/TextField.js

```diff
@@ -0,0 +1,266 @@
+import { MailIcon, XIcon, CheckIcon } from "@heroicons/react/outline";
+import { labelWeights, iconAlignment } from "./TextField.levers";
+
+/**
+ * Text Input  Component
+ *
+ */
+export const TextField = (
+  {
+    className,
+    id,
+    label,
+    fontWeight,
+    helperText,
+    placeholder,
+    icon,
+    type,
+    maxLength,
+    minLength,
+    errorText,
+    validText,
+    isError,
+    isValid,
+    isIcon,
+    iconPosition,
+    isDisabled,
+    isRequired,
+    showCharCount,
+  },
+  props
+) => {
+  // Defaults & Variables.
+  // ---------------------------------------------------------------------------
+  let weight = "regular";
+  let disabledStyle = "";
+  let errorLabelStyle = "";
+  let errorInputStyle = "";
+  let iconPositionStyle =
+    "su-absolute su-bottom-0 su-left-0 su-ml-19 su-mt-19 su-mb-20";
+  let iconInputStyle = "";
+
+  // Default Icon — Mail Envelope
+  const defaultIcon = <MailIcon />;
+
+  // Levers
+  // ---------------------------------------------------------------------------
+
+  // Font Weight
+  if (fontWeight && fontWeight in labelWeights) {
+    weight = labelWeights[fontWeight];
+  }
+
+  if (isDisabled) {
+    disabledStyle = "su-bg-black-10";
+  }
+
+  if (isError) {
+    errorLabelStyle = "su-text-digital-red";
+    errorInputStyle = "su-border-digital-red";
+  }
+
+  if (isIcon && iconPosition === "left") {
+    iconInputStyle = "su-pl-58";
+  }
+
+  if (iconPosition === "right") {
+    iconPositionStyle =
+      "su-absolute su-bottom-0 su-right-0 su-mr-19 su-mt-19 su-mb-20";
+  }
+
+  const inputIcon = icon ?? defaultIcon;
+
+  // Handle Character Counter
+  // ---------------------------------------------------------------------------
+  const [charCount, setCharCount] = useState(0);
+  const handleOnChange = (e) => {
+    setCharCount(e.target.value.length);
+  };
+
+  return (
+    <div className={className}>
+      <label
+        htmlFor={id}
+        className={dcnb("su-label su-mb-5 su-text-18", weight, errorLabelStyle)}
+      >
+        {isRequired ? "*" : ""}
+        {label}
+      </label>
+      {helperText && (
+        <p className="su-text-cool-grey su-mb-5 su-text-18 su-sans">
+          {helperText}
+        </p>
+      )}
+      <div className="su-relative su-w-300">
+        {isIcon ? (
+          <span className={dcnb("su-w-30", iconPositionStyle)}>
+            {inputIcon}
+          </span>
+        ) : (
+          ""
+        )}
+        <input
+          id={id}
+          className={dcnb(
+            "su-input su-border-b-2 su-rounded su-pt-19 su-pb-20 su-px-19 su-w-full",
+            disabledStyle,
+            errorInputStyle,
+            iconInputStyle
+          )}
+          type={type}
+          placeholder={placeholder}
+          maxLength={maxLength ?? null}
+          minLength={minLength ?? null}
+          disabled={isDisabled}
+          required={isRequired}
+          onChange={handleOnChange}
+          {...props}
```

Comment:
Assuming that the consumer of this component passes an `onChange` handler wouldn't the above `handleOnChange` get overwritten?